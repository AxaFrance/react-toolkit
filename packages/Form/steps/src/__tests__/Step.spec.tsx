import React from 'react';
import { render } from '@testing-library/react';
import { Steps, Step, StepBase } from '../index';

describe('<Steps>', () => {
  it('renders Steps correctly', () => {
    const { asFragment } = render(
      <Steps>
        <Step
          id="id1"
          href="/etape1"
          onClick={jest.fn()}
          number="1"
          mode="link"
          title="Previous step"
        />
        <StepBase id="idf4" title="Un titre">
          <a className="af-steps-list-stepLabel" href="/#" onClick={jest.fn()}>
            <span className="af-steps-list-stepNumber">
              <i className="glyphicon glyphicon-ok" />
            </span>
            <span className="af-steps-list-stepTitle">Custom</span>
          </a>
        </StepBase>
        <Step id="id5" number="5" title="Final step" mode="disabled" />
      </Steps>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
