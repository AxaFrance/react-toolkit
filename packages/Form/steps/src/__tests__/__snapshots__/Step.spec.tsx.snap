// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Steps> renders Steps correctly 1`] = `
<DocumentFragment>
  <div
    class="af-steps"
  >
    <ol
      class="af-steps-list"
    >
      <li
        class="past af-steps-list-step number"
        title="Previous step"
      >
        <a
          class="af-steps-list-stepLabel"
          href="/etape1"
        >
          <div
            class="af-steps-list-stepNumber"
          >
            1
          </div>
          <div
            class="af-steps-list-stepTitle"
          >
            Previous step
          </div>
        </a>
      </li>
      <li
        class="af-steps-list-step"
        title="Un titre"
      >
        <a
          class="af-steps-list-stepLabel"
          href="#"
        >
          <span
            class="af-steps-list-stepNumber"
          >
            <i
              class="glyphicon glyphicon-ok"
            />
          </span>
          <span
            class="af-steps-list-stepTitle"
          >
            Custom
          </span>
        </a>
      </li>
      <li
        class="af-steps-list-step number"
        title="Final step"
      >
        <div
          class="af-steps-list-stepLabel"
        >
          <div
            class="af-steps-list-stepNumber"
          >
            5
          </div>
          <div
            class="af-steps-list-stepTitle"
          >
            Final step
          </div>
        </div>
      </li>
    </ol>
  </div>
</DocumentFragment>
`;
